src/posts/en/javascript-iterables-and-iterators.md

```diff
@@ -0,0 +1,294 @@
+<!--
+layout: post
+title: JavaScript iterables and iterators
+date: 2015-09-13T01:41:02.428Z
+comments: true
+published: true
+keywords: iterables, iterators, ES2015
+description: Understanding JavaScript ES2015 Iterables and Iterators
+categories: iterables, iterators, ES2015, articles
+authorName: Fabrício S. Matté
+authorLink: https://twitter.com/Ult_Combo
+authorDescription: ECMAScript enthusiast, open source addict and Web Platform lover.
+authorPicture: https://pbs.twimg.com/profile_images/490627147963187200/2BiH3pv4.png
+-->
+ECMAScript 2015 (ES6) introduces two new distinct, yet closely related, concepts: **iterables** and **iterators**.<br>
+Now you will truly learn how important these concepts are and how to make good use of them.
+<!--more-->
+
+## Iterables
+
+As the name implies, iterable objects are objects that implement the [Iterable interface](http://www.ecma-international.org/ecma-262/6.0/index.html#sec-iterable-interface). That is, iterable objects expose a default iteration method, allowing them to define or customize their iteration behavior.
+
+Let's see some examples:
+
+```javascript
+let iterable = [1, 2, 3];
+for (let item of iterable) {
+	console.log(item); // 1, 2, 3
+}
+
+let iterable2 = new Set([4, 5, 6]);
+for (let item of iterable2) {
+	console.log(item); // 4, 5, 6
+}
+
+let iterable3 = '789';
+for (let item of iterable3) {
+	console.log(item); // '7', '8', '9'
+}
+```
+
+As you can see, the [for-of](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Statements/for...of) statement accepts any iterable object, thus providing an uniform iteration syntax to completely distinct data structures by making use of the Iterable interface they implement.
+
+### So what exactly is this Iterable interface you speak of?
+
+It is simple, really. Any object that contains a `[Symbol.iterator]` method is an iterable object.<br>
+In case you are unfamilar with symbols, they are, basically, objects that can be used as property keys.<br>
```

Comment:
unfamilar > unfamiliar

And there are perhaps too many commas. I would use:
"In case you are unfamiliar with symbols, they are basically objects that can be used as property keys"